
```{r setup, echo =FALSE}
.libPaths('C:\\Users\\zuendj\\Documents\\R4Libs')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "P:/Shared_documents/Serafina PlÃ¼ss/20230320_sequencing_results/Janina", cho = TRUE, warning = FALSE, message = FALSE)

# library("plyr")
# install.packages("vctrs")
# install.packages("vegan")
# install.packages("cli")

# library(vctrs)
# library(vegan)
# install.packages("ggplot2")
#library(ggplot2)
#install.packages('tinytex')
#install.packages('rmarkdown')
#install.packages("rlang")
#install.packages("nlme")
#install.packages("mgcv")
#install.packages("ape")
#library(nlme)
# install.packages("remotes")
# remotes::install_github("mikemc/speedyseq")
# install.packages("BiocManager")
# BiocManager::install("DESeq2")
#install.packages("fs")
# install.packages('devtools')
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
#unload("mia")
# install.packages("GUniFrac")

# install.packages(
#   "microViz",
#   repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos"))
# )
# devtools::install_github("tidyverse/tidyverse")
 # library(tidyverse)
  # library(openxlsx)
# # install.packages("ggalluvial")
# library(ggalluvial)
 # library(ggh4x)
# library(GUniFrac)
 # library(rstatix)
# library(tinytex)
# library(rmarkdown)
# library(rlang) 
# library(tidyverse)
# library(microViz)
# library(dplyr)
# library(mgcv)
# library(ape)
# library(phyloseq)
# library(speedyseq)
# library(DESeq2)
 # library(plyr)
# library(fs)
# library(devtools)
# library(pairwiseAdonis)
# library(ggpubr)

# install.packages("dplyr")
# install.packages("data.table")
# devtools::install_github("vmikk/metagMisc")
#  
# library(metagMisc)


```

```{r, echo =FALSE}
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")

#get metadata file: load this function
sample.data.frame <- function(ps) {
  return(as(phyloseq::sample_data(ps), "data.frame"))}
```

## 1. Get the phyloseq object 
```{r}
 unloadNamespace("miaViz")
 unloadNamespace("mia")

phyloseq <-readRDS("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/phyloseq.RDS")
phyloseq%>%
  phyloseq_get_strains()->phyloseq
```



```{r}
hplc<-read.xlsx("C:/Users/zuendj/Desktop/03_data R/wp2_complex/input/20230613_hplc_list_results_stress.xlsx")
hplc%>%
  subset(!sample_name %in% NA)->hplc

seq<-read.xlsx("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/mapping_file.xlsx")

meta<-left_join(seq, hplc[,c(1:12)])
write.xlsx(meta, "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta_new.xlsx")
```

### Get and update metadata file
```{r, echo=FALSE}

phyloseq%>%
  physeq_add_metadata(physeq = .,
                      metadata = "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta.xlsx" %>%
                        readxl::read_xlsx(),
                      sample_column = "sample_name") -> phyloseq

as.data.frame(sample_data(phyloseq))->meta
## remove donor 2 from data as very low sample size
# phyloseq %>%
#   subset_samples(!donor_name %in% "D1")->phyloseq



```

```{r}
ps_strain_filt <- phyloseq

# threshold in %
threshold = 0.1

# filter per sample to remove all ASV lower in abundance than threshold (%)
otu_table(ps_strain_filt) <- otu_table(ps_strain_filt) %>%
  as.data.frame() %>%
  dplyr:: mutate(across(everything(), ~ ifelse(. <= sum(.) * (threshold/100), 0, .))) %>% otu_table(., taxa_are_rows = TRUE)

# remove all ASVs that are not present in any sample
ps_strain_filt <- ps_strain_filt %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE)

noNA = !is.na(tax_table(ps_strain_filt)[,"Genus"]) & !is.na(tax_table(ps_strain_filt)[,"Species"])
tax_table(ps_strain_filt)[noNA][,"Species"] = paste(tax_table(ps_strain_filt)[noNA][,"Genus"], tax_table(ps_strain_filt)[noNA][,"Species"])
ps_strain_filt%>%
  tax_fix()->ps_strain_filt


```



#Differtial abunance
```{r, echo=FALSE}
#BiocManager::install("microbiome/mia")

 library(mia)
 library(ALDEx2)
```

```{r}
ox <- makeTreeSummarizedExperimentFromPhyloseq(ps_strain_filt) 
```


```{r}
ox_fam <- ox %>% agglomerateByRank(rank = "Family")
ox_gen <- ox %>% agglomerateByRank(rank = "Genus")
ox_species <- ox %>% agglomerateByRank(rank = "Species") 
```

###Aldex: NO Prevalence filter-> filter later with abundance: Genus

```{r}
set.seed(123)

oxygen <-ox_gen[ , ox_gen$stress %in% c("O2") ]
stress_level_cond <-c("control", "low","median", "high", "max")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_oxygen_genus <-data.frame(matrix(ncol=15, nrow=0))

for (i in (1:length(stress_level_cond))){
  # ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) ]
  
   ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) & !oxygen$donor_name %in% c("D1", "D5")]
  
  
    for (j in (1:length(time_cond))){
  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
    result_name <- paste(stress_level_cond[i])
   
      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$incubation, 
     mc.samples=128)

      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- paste(time_cond[j])
      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_oxygen_genus <- rbind(all_oxygen_genus, result)
      }}
all_oxygen_genus %>%
  subset(V15 %in% "stress")->out

unique(out$Family)

out[,c(1,3,6,8,10, 14,15,16)]%>%
  subset(wi.ep < 0.05)
```


###Aldex: H2O2
###genus level
#atm i take wi.ep => but I should correct for multiple testing in this case => think about how to do it => no idea what wi.eBH is exactly
```{r}
set.seed(123)
h2o2 <-ox_gen[ , ox_gen$stress %in% c("H2O2")]

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_h2o2_genus <-data.frame(matrix(ncol=15, nrow=0))
j=0 
count=0
donors <- list(
  donor_all = character(0),
  donor_1 = c("D1"),
  donor_2 = c("D1", "D5"),
  donor_3 = c("D1", "D2", "D3", "D5")
)

# Create a vector to store the combined donor conditions
donor_cond <- character(0)

# for (p in donors) {
#   donor_cond <- c(p)
#   print(donor_cond)
   for (i in (1:length(stress_level_cond))){
  for (j in (1:length(time_cond))){
  ox_sub <-h2o2[ , h2o2$stress_level %in% c(stress_level_cond[i], "control") & 
                   !(h2o2$donor_name %in% c("D1"))&
                   (h2o2$time %in% c(time_cond[j]))]

 

  
      ox_sub
      result_name <- paste(stress_level_cond[i])

      x <- aldex.clr(
      reads = assay(ox_sub),
     ox_sub$stress_level, mc.samples=128)
      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- time_cond[j]

      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_h2o2_genus <- rbind(all_h2o2_genus, result)
      

 }}

all_h2o2_genus %>%
  subset(V15 %in% "stress")%>%
  filter(wi.ep < 0.05)->out


uni<-(unique(out$Family)) 
print(uni)

# }

all_h2o2_genus %>%
# out%>%
  subset(wi.ep <0.05 & V15 %in% "stress")-> sig_taxa

all_h2o2_genus[, c("wi.ep", "wi.eBH", "diff.btw" ,"V14", "V15", "Family")] %>%
  # subset(Family %in% "Lactobacillales Order")%>%
  subset(wi.ep <0.05)%>%
  filter(diff.btw > 0)


```


```{r}

unloadNamespace("miaViz")
unloadNamespace("mia")
```

