## Differential abundance analysis

```{r setup, echo =FALSE}
.libPaths('C:\\Users\\zuendj\\Documents\\R4Libs')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "P:/Shared_documents/Serafina PlÃ¼ss/20230320_sequencing_results/Janina", cho = TRUE, warning = FALSE, message = FALSE)

```



## 1. Get the phyloseq object 
```{r}
 unloadNamespace("miaViz")
 unloadNamespace("mia")

phyloseq_rare <-readRDS("C:/Users/zuendj/Desktop/03_data R/wp2_complex/input/phyloseq_nonrare.RDS")


```

# Differtial abunance
```{r, echo=FALSE}
#BiocManager::install("microbiome/mia")

 library(mia)
 library(ALDEx2)
```

```{r}
ox <- makeTreeSummarizedExperimentFromPhyloseq(phyloseq_rare) 
```


```{r}
ox_gen <- ox %>% agglomerateByRank(rank = "Genus")
```

###Aldex: NO Prevalence filter-> filter later with abundance: Genus

```{r}
set.seed(123)

oxygen <-ox_gen[ , ox_gen$stress %in% c("O2") ]


stress_level_cond <-c("control", "low", "median", "high","max")
time_cond<-c("stress", "post_stress_1")
all_oxygen_genus <-data.frame(matrix(ncol=15, nrow=0))

for (i in (1:length(stress_level_cond))){
  # ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) ]
  
   ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) ]
  
  
    for (j in (1:length(time_cond))){
  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
    result_name <- paste(stress_level_cond[i])
   
      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$incubation, 
     mc.samples=128)

      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- paste(time_cond[j])
      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_oxygen_genus <- rbind(all_oxygen_genus, result)
      }}
all_oxygen_genus ->out



 

out[,c(1,2,3,4,6,8,10, 14,15,16)]%>%

   subset(wi.ep < 0.05)%>%
  subset(V15 %in% "stress")->o

filter_o2<-unique(o$Family)
```

```{r}
# library(scales)

# filter_o2<-unique(o$Family)

out[,c(1,2,3,4,6,8,10, 14,15,16)]%>%
   subset( diff.btw > 0.75 )%>%
 
   # subset(wi.ep < 0.05)%>%
  subset(V15 %in% "stress")->o


out[,c(1,2,3,4,6,8,10, 14,15,16)]%>%
   subset( diff.btw < -0.75 | wi.ep < 0.05)%>%
   subset(!Family %in% "Erysipelotrichaceae UCG-003")%>%subset(V15 %in% "stress")->oo


dir<-rbind(data.frame(Family = unique(oo$Family),
          direction = rep ("decreased", length(unique(oo$Family)))),
data.frame(Family = unique(o$Family),
          direction = rep ("increased", length(unique(o$Family)))))

out%>%
   mutate(diff.btw = (-1) * diff.btw)%>%
  subset(Family %in% c(filter_o2))%>%
  mutate(Family = reorder(Family, diff.btw))%>%
  mutate(we.ep = ifelse(we.ep < 0.001 , "***", we.ep))%>%
  mutate(we.ep = ifelse(we.ep < 0.05 & we.ep >= 0.01, "*", we.ep))%>%
  mutate(we.ep = ifelse(we.ep < 0.01 & we.ep >= 0.001, "**", we.ep))%>%
 
  mutate(we.ep = ifelse(!we.ep == c("*") & !we.ep == "**" & !we.ep == "***", NA, we.ep))%>%
  # mutate(V14 = ifelse(V14  == "low","0.22 mM", V14))%>%
  # mutate(V14 = ifelse(V14  == "median","0.71 mM", V14))%>%
  #   mutate( V14 = ifelse(V14 == "high", "2.3 mM", V14))%>%
   mutate(V14 = factor(V14, levels = c("control", "low",  "median", "high", "max")))%>%
  mutate(V15 = factor( V15, levels=c("stress", "post_stress_1")))%>%


  ggplot(aes(x=V15, y=Family, fill=diff.btw))+
  geom_tile()+
   facet_grid(cols=vars(V14), scales = "free", space="free")+
 scale_fill_gradientn(colors = c( "coral", "white", "cyan4"),  na.value= "grey",
                      values = rescale(c(-3, 0, 3)), limits=c(-3, 3),
                       name="Clr-difference")+
   theme(text = element_text(size=15),
         axis.title = element_blank(),
        axis.text.x = element_text(angle = 90,  vjust=0.5, hjust=1),
        axis.text.y = element_text( face="italic"), 
        panel.border=element_rect(fill=NA), 
        panel.background = element_rect(fill="white"),
        legend.text.align = 0, 
        strip.text.y = element_blank(),
        strip.background = element_blank(),
        panel.spacing = unit(0, "lines"))+
    scale_x_discrete(labels=c( 'stress' = "Stress", 'post_stress_1' = "Post-stress"))+
  geom_text(aes(label=we.ep), size=5)

ggsave(filename = paste("diff_abundo2.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/final",
       width = 15,
       height = 6,
       units = c("cm"))

```




###Aldex: H2O2
###genus level
#atm i take wi.ep => but I should correct for multiple testing in this case => think about how to do it => no idea what wi.eBH is exactly
```{r}
set.seed(123)
h2o2 <-ox_gen[ , ox_gen$stress %in% c("H2O2")]

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1")
all_h2o2_genus <-data.frame(matrix(ncol=15, nrow=0))
j=0 
count=0
donors <- list(
  donor_all = character(0),
  donor_1 = c("D1"),
  donor_2 = c("D1", "D5"),
  donor_3 = c("D1", "D2", "D3", "D5")
)

# Create a vector to store the combined donor conditions
donor_cond <- character(0)

# for (p in donors) {
#   donor_cond <- c(p)
#   print(donor_cond)
   for (i in (1:length(stress_level_cond))){
  for (j in (1:length(time_cond))){
  ox_sub <-h2o2[ , h2o2$stress_level %in% c(stress_level_cond[i], "control") & 
                   (h2o2$time %in% c(time_cond[j]))]


      ox_sub
      result_name <- paste(stress_level_cond[i])

      x <- aldex.clr(
      reads = assay(ox_sub),
     ox_sub$stress_level, mc.samples=128)
      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- time_cond[j]

      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_h2o2_genus <- rbind(all_h2o2_genus, result)
      

 }}

all_h2o2_genus %>%
  # subset(V15 %in% "stress")%>%
  filter(we.ep < 0.05)->out


uni<-(unique(out$Family)) 
print(uni)

# }

all_h2o2_genus %>%
# out%>%
  subset(we.ep <0.05  &
           V15 %in% "stress")-> sig_taxa

all_h2o2_genus[, c("we.ep", "wi.eBH", "diff.btw" ,"V14", "V15", "Family")] %>%
  # subset(Family %in% "Bacteroides")
  
  # 
    mutate(we.ep = we.ep *3)%>%
  subset(we.ep <0.05)->sig_taxa

 filter<- unique(sig_taxa$Family)
 
```


```{r}

unloadNamespace("miaViz")
unloadNamespace("mia")

 library(scales)
filter_decreased<-c("Bacteroides", "Fusicatenibacter", "Anaerostipes", 
               "Blautia", "Agathobacter",
             "Dorea", "Faecalibacterium", 
             "Lachnoclostridium", "Lachnospiraceae UCG-004", "Lachnospiraceae Family",
             "[Eubacterium] hallii group",
             "Parabacteroides")

filter_increased<-c("Enterobacteriaceae Family", "Escherichia-Shigella","Bifidobacterium")
# , "Streptococcus"
length(filter_decreased)

dir<-rbind(data.frame(Family = filter_decreased,
          direction = rep ("decreased", length(filter_decreased))),
data.frame(Family = filter_increased,
          direction = rep ("increased", length(filter_increased))))


left_join(all_h2o2_genus[, c("we.ep", "wi.ep", "diff.btw" ,"V14", "V15", "Family", "rab.all")], dir)%>%
  mutate(we.ep = we.ep *3)%>%
   mutate(we.ep = ifelse(we.ep < 0.001 , "***", we.ep))%>%
  mutate(we.ep = ifelse(we.ep < 0.05 & we.ep >= 0.01, "*", we.ep))%>%
  mutate(we.ep = ifelse(we.ep < 0.01 & we.ep >= 0.001, "**", we.ep))%>%
 
  mutate(we.ep = ifelse(!we.ep == c("*") & !we.ep == "**" & !we.ep == "***", NA, we.ep))%>%
  mutate(V14 = ifelse(V14  == "low","0.22 mM", V14))%>%
  mutate(V14 = ifelse(V14  == "median","0.71 mM", V14))%>%
    mutate( V14 = ifelse(V14 == "high", "2.3 mM", V14))%>%
  mutate(V14 = factor(V14, levels = c("0.22 mM", "0.71 mM",  "2.3 mM")))%>%
  mutate(V15 = factor( V15, levels=c("stress", "post_stress_1")))%>%
  subset(Family %in% c(filter_decreased, filter_increased))%>%
  mutate(Family = factor(Family))%>%
  mutate(Family = reorder(Family, rab.all))%>%
  ggplot(aes(x=V15, y=Family, fill=diff.btw))+
  geom_tile()+
  facet_grid(cols=vars(V14), rows=vars(direction), scales = "free", space="free")+
 scale_fill_gradientn(colors = c( "coral", "white", "cyan4"),  na.value= "grey", values = rescale(c(-12, 0, 12)), limits=c(-12, 12), name="Clr-difference")+
   theme(text = element_text(size=15),
         axis.title = element_blank(),
        axis.text.x = element_text(angle = 90,  vjust=0.5, hjust=1),
        axis.text.y = element_text( face="italic"), 
        panel.border=element_rect(fill=NA), 
        panel.background = element_rect(fill="white"),
        legend.text.align = 0, 
        strip.text.y = element_blank(),
        strip.background = element_blank(),
        panel.spacing = unit(0, "lines"))+
    scale_x_discrete(labels=c( 'stress' = "Stress", 'post_stress_1' = "Post-stress"))+
  geom_text(aes(label=we.ep), size=5)

ggsave(filename = paste("diff_abundh2o2.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/final",
       width = 16,
       height = 13,
       units = c("cm"))
  
```

```{r}
unloadNamespace("miaViz")
unloadNamespace("mia")
```

