
```{r setup, echo =FALSE}
.libPaths('C:\\Users\\zuendj\\Documents\\R4Libs')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "P:/Shared_documents/Serafina PlÃ¼ss/20230320_sequencing_results/Janina", cho = TRUE, warning = FALSE, message = FALSE)


# devtools::install_github("tidyverse/tidyverse")
 library(tidyverse)
 library(openxlsx)
 library(ggh4x)
 library(microViz)

```

```{r, echo =FALSE}
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")

#get metadata file: load this function
sample.data.frame <- function(ps) {
  return(as(phyloseq::sample_data(ps), "data.frame"))}


```

## 1. Get the phyloseq object 
```{r}
unloadNamespace("miaViz")
unloadNamespace("mia")

phyloseq <-readRDS("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/phyloseq.RDS")
phyloseq%>%
  phyloseq_get_strains()->phyloseq
```


```{r}
hplc<-read.xlsx("C:/Users/zuendj/Desktop/03_data R/wp2_complex/input/20230613_hplc_list_results_stress.xlsx")
hplc%>%
  subset(!sample_name %in% NA)->hplc

seq<-read.xlsx("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/mapping_file.xlsx")

meta<-left_join(seq, hplc[,c(1:12)])
write.xlsx(meta, "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta_new.xlsx")
```

### Get and update metadata file
```{r, echo=FALSE}

phyloseq%>%
  physeq_add_metadata(physeq = .,
                      metadata = "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta.xlsx" %>%
                        readxl::read_xlsx(),
                      sample_column = "sample_name") -> phyloseq

as.data.frame(sample_data(phyloseq))->meta
## remove donor 2 from data as very low sample size
# phyloseq %>%
#   subset_samples(!donor_name %in% "D1")->phyloseq



```

```{r}
ps_strain_filt <- phyloseq

# threshold in %
threshold = 0.1

# filter per sample to remove all ASV lower in abundance than threshold (%)
otu_table(ps_strain_filt) <- otu_table(ps_strain_filt) %>%
  as.data.frame() %>%
  dplyr:: mutate(across(everything(), ~ ifelse(. <= sum(.) * (threshold/100), 0, .))) %>% otu_table(., taxa_are_rows = TRUE)

# remove all ASVs that are not present in any sample
ps_strain_filt <- ps_strain_filt %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE)
```

#Rarefying
```{r}
# phyloseq%>%
ps_strain_filt%>%
  phyloseq_check_lib_size(data_color = "donor_name",
                          data_facet = NULL,
                          nreads_display = 3165,
                       first_n = nsamples(phyloseq)) -> lib

ps_strain_filt%>%
# phyloseq %>%
  rarefy_even_depth(rngseed = 123,
                    sample.size = 2900
                    # sample.size = 5000
                    ) -> phyloseq_rare




df<-lib$df
```
#Fix tax table
```{r}
# phylo_C@otu_table
# phylo_C@tax_table
noNA = !is.na(tax_table(phyloseq_rare)[,"Genus"]) & !is.na(tax_table(phyloseq_rare)[,"Species"])
tax_table(phyloseq_rare)[noNA][,"Species"] = paste(tax_table(phyloseq_rare)[noNA][,"Genus"], tax_table(phyloseq_rare)[noNA][,"Species"])
phyloseq_rare%>%
  tax_fix()->phyloseq_rare


noNA = !is.na(tax_table(ps_strain_filt)[,"Genus"]) & !is.na(tax_table(ps_strain_filt)[,"Species"])
tax_table(ps_strain_filt)[noNA][,"Species"] = paste(tax_table(ps_strain_filt)[noNA][,"Genus"], tax_table(ps_strain_filt)[noNA][,"Species"])
ps_strain_filt%>%
  tax_fix()->ps_strain_filt

```

```{r}
myPal <- tax_palette(
  data = phyloseq_rare, rank = "Family", n = 10, pal = "greenArmytage",
  add = c(Other = "white")
)


myPal["Lachnospiraceae"] <- "lightblue"
myPal["Bacteroidaceae"] <- "steelblue"
myPal["Enterobacteriaceae"] <- "yellowgreen"
myPal["Prevotellaceae"] <- "forestgreen"
myPal["Ruminococcaceae"] <- "lightpink1"

myPal["Bifidobacteriaceae"] <- "red3"
myPal["Oscillospiraceae"] <- "sienna1"
myPal["Streptococcaceae"] <- "orange"
myPal["Sutterellaceae"] <- "khaki1"
myPal["Erysipelatoclostridiaceae"] <- "orchid4"
myPal["Erysipelotrichaceae"] <- "orchid2"
myPal["Christensenellaceae"] <- "orchid3"
myPal["Tannerellaceae"] <- "lightpink3"
myPal["Coriobacteriaceae"] <- "sienna"
myPal["Enterococcaceae"] <- "red4"
myPal["Veillonellaceae"] <- "lightblue3"
myPal["Other"] <- "lightgrey"
myPal["Peptostreptococcaceae"] <- "khaki3"
myPal["Clostridiaceae"]<-"khaki"
myPal["Lactobacillaceae"]<-"darkseagreen2"
myPal["Akkermansiaceae"]<-"darkseagreen4"

```

#### compoistion feces ####
```{r}
phyloseq_rare%>%
  subset_samples(incubation %in% "feces")%>%
  ps_arrange(time)%>%
      microViz::comp_barplot(
    tax_level = "Family",
    label = "donor_name",
    tax_order = sum, 
    sample_order = "bray",
    n_taxa = 30,
    bar_width = 0.9,
    tax_transform_for_plot = "compositional") +
    ylab("Proportion") + 
    xlab( "")-> p_hist

p_hist + 
  # facet_grid(cols=vars(stress_level), drop = TRUE, scales = "free_x", space="free")+
  theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
        axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
        axis.text.y= element_text(size = 20), 
        axis.title.y = element_text(size=20), 
        axis.title.x = element_text(size=20),
        legend.text = element_text(size = 15, vjust=1, hjust=0),
        legend.title = element_text(size= 20),
        plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) + 
  scale_y_continuous(breaks=c(0.25, 0.5, 0.75))+
  ggtitle("Taxonomic distribution - feces")-> p_hist

p_hist

ggsave(filename = "bar_feces.jpeg",
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 30,
       height = 14,
       units = c("cm"))
  
```
# composition of pre-cultures
```{r}
phyloseq_rare%>%
  subset_samples(time %in% c("pre-stress", "feces"))%>%
  ps_arrange(time)%>%
      microViz::comp_barplot(
    tax_level = "Family",
    label = "donor_name",
    tax_order = sum, 
    sample_order = "bray",
    n_taxa = 30,
    bar_width = 0.9,
    tax_transform_for_plot = "compositional") +
    ylab("Proportion") + 
    xlab( "")-> p_hist

p_hist + 
  # facet_grid(cols=vars(donor_name), drop = TRUE, scales = "free_x", space="free")+
  theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
        axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
        axis.text.y= element_text(size = 20), 
        axis.title.y = element_text(size=20), 
        axis.title.x = element_text(size=20),
        legend.text = element_text(size = 15, vjust=1, hjust=0),
        legend.title = element_text(size= 20),
        plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) + 
  scale_y_continuous(breaks=c(0.25, 0.5, 0.75)) +
  ggtitle("Taxonomic distribution - pre-cultures")-> p_hist

p_hist


ggsave(filename = "bar_pre-cultures.jpeg",
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 40,
       height = 14,
       units = c("cm"))
```


#### Taxa bar plots ####
##Family level -H2O2
```{r}
# 
# donors_loop<-c("D1","D2", "D3", "D4", "D5", "D6", "D7", "D8")
# stress_level_cond<-c("control", "low", "median", "high")
# 
# # tax_palette_plot(myPal)
# 
# phyloseq_rare%>%
#   subset_samples(time %in% c("feces", "pre-stress" , "stress", "post_stress_1", "post_stress_2" ))->phyloseq_bar
# 
# phyloseq_bar%>%
#   sample.data.frame()->meta
# 
# meta%>%
#   mutate(stress_level = ifelse(time == "pre-stress", "pre", stress_level))%>%
#   mutate(time= factor(time, levels=c("feces", "pre-stress", "stress", "post_stress_1", "post_stress_2")))->meta
#                                           
# #update the metadata file
# sample_data(phyloseq_bar)<-meta
# 
# 
# 
# for (i in 1:length(donors_loop)){
#   for (j in 1:length(stress_level_cond)){
# 
# phyloseq_bar%>% 
#     subset_samples(donor_name %in% c(donors_loop[i]))%>%
#      subset_samples(stress %in% c("H2O2","control", "feces"))%>%
#       subset_samples(stress_level %in% c(stress_level_cond[j],"pre", "feces"))%>%
#      ps_arrange(time)%>%
#       microViz::comp_barplot(
#     tax_level = "Family",
#     label = "time",
#     tax_order = sum, 
#     sample_order = "default",
#     n_taxa = 10,
#     bar_width = 0.9,
#     tax_transform_for_plot = "compositional",
#       palette= myPal) +
#     ylab("Proportion") + 
#     xlab( "")-> p_hist
# 
#  
# p_hist + 
#   # facet_grid(cols=vars(stress_level), drop = TRUE, scales = "free_x", space="free")+
#   theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
#         axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
#         axis.text.y= element_text(size = 20), 
#         axis.title.y = element_text(size=20), 
#         axis.title.x = element_text(size=20),
#         legend.text = element_text(size = 15, vjust=1, hjust=0),
#         legend.title = element_text(size= 20),
#         plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) + 
#   scale_y_continuous(breaks=c(0.25, 0.5, 0.75))+ 
#   ggtitle(paste(donors_loop[i], "-", stress_level_cond[j],  " stress")) -> p_hist
# 
# p_hist
# 
# 
# 
# x<-paste0(donors_loop[i],stress_level_cond[j],"H2O2_stress_box.jpeg")
# ggsave(filename = x,
#        path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/H2O2_bar",
#        width = 14,
#        height = 14,
#        units = c("cm"))
# 
# }}
```

##Strain level H2O2
```{r}

donors_loop<-c("D1","D2", "D3", "D4", "D5", "D6", "D7", "D8")
stress_level_cond<-c("control", "low", "median", "high")





phyloseq_rare%>%
  subset_samples(time %in% c("feces", "pre-stress" , "stress", "post_stress_1", "post_stress_2" ))->phyloseq_bar

phyloseq_bar%>%
  sample.data.frame()->meta



meta%>%
  mutate(stress_level = ifelse(time == "pre-stress", "pre", stress_level))%>%
  mutate(time= factor(time, levels=c("feces", "pre-stress", "stress", "post_stress_1", "post_stress_2")))->meta





#update the metadata file
sample_data(phyloseq_bar)<-meta



for (i in 1:length(donors_loop)){
  for (j in 1:length(stress_level_cond)){

phyloseq_bar%>%
    subset_samples(donor_name %in% c(donors_loop[i]))%>%
     subset_samples(stress %in% c("H2O2","control", "feces"))%>%
      subset_samples(stress_level %in% c(stress_level_cond[j],"pre", "feces"))%>%
     ps_arrange(time)%>%
      microViz::comp_barplot(
    tax_level = "Genus",
    label = "time",
    tax_order = sum,
    sample_order = "default",
    n_taxa = 20,
    bar_width = 0.9,
    tax_transform_for_plot = "compositional"
      ) +
    ylab("Proportion") +
    xlab( "")-> p_hist


p_hist +
  # facet_grid(cols=vars(stress_level), drop = TRUE, scales = "free_x", space="free")+
  theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
        axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
        axis.text.y= element_text(size = 20),
        axis.title.y = element_text(size=20),
        axis.title.x = element_text(size=20),
        legend.text = element_text(size = 15, vjust=1, hjust=0),
        legend.title = element_text(size= 20),
        plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) +
  scale_y_continuous(breaks=c(0.25, 0.5, 0.75))+
  ggtitle(paste(donors_loop[i], "-", stress_level_cond[j],  " stress")) -> p_hist

p_hist



x<-paste0(donors_loop[i],stress_level_cond[j],"H2O2_stress_box.jpeg")
ggsave(filename = x,
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/H2O2_bar/Genus",
       width = 24,
       height = 14,
       units = c("cm"))

}}


```
##Family level -O2
```{r}

# donors_loop<-c("D1","D2", "D3", "D4", "D5", "D6", "D7", "D8")
# stress_level_cond<-c("control", "low", "median", "high", "max")
# 
# 
# phyloseq_rare%>%
#   subset_samples(time %in% c("feces", "pre-stress" , "stress", "post_stress_1", "post_stress_2" ))->phyloseq_bar
# 
# phyloseq_bar%>%
#   sample.data.frame()->meta
# 
# 
# 
# meta%>%
#   mutate(stress_level = ifelse(time == "pre-stress", "pre", stress_level))%>%
#   mutate(time= factor(time, levels=c("feces", "pre-stress", "stress", "post_stress_1", "post_stress_2")))->meta
#                                             
# #update the metadata file
# sample_data(phyloseq_bar)<-meta
# 
# 
# 
# for (i in 1:length(donors_loop)){
#   for (j in 1:length(stress_level_cond)){
# 
# phyloseq_bar%>% 
#     subset_samples(donor_name %in% c(donors_loop[i]))%>%
#      subset_samples(stress %in% c("O2","control", "feces"))%>%
#       subset_samples(stress_level %in% c(stress_level_cond[j],"pre", "feces"))%>%
#       subset_samples(incubation %in% "aerobe" | (stress_level %in% c("pre", "feces")))%>%
#      ps_arrange(time)%>%
#       microViz::comp_barplot(
#     tax_level = "Family",
#     label = "time",
#     tax_order = sum, 
#     sample_order = "default",
#     n_taxa = 10,
#     bar_width = 0.9,
#     tax_transform_for_plot = "compositional",
#       palette= myPal) +
#     ylab("Proportion") + 
#     xlab( "")-> p_hist
# 
#  
# p_hist + 
#   # facet_grid(cols=vars(stress_level), drop = TRUE, scales = "free_x", space="free")+
#   theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
#         axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
#         axis.text.y= element_text(size = 20), 
#         axis.title.y = element_text(size=20), 
#         axis.title.x = element_text(size=20),
#         legend.text = element_text(size = 15, vjust=1, hjust=0),
#         legend.title = element_text(size= 20),
#         plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) + 
#   scale_y_continuous(breaks=c(0.25, 0.5, 0.75))+ 
#   ggtitle(paste(donors_loop[i], "-", stress_level_cond[j],  " stress")) -> p_hist
# 
# p_hist
# 
# 
# 
# x<-paste0(donors_loop[i],stress_level_cond[j],"H2O2_stress_box.jpeg")
# ggsave(filename = x,
#        path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/O2_bar/aerobe",
#        width = 14,
#        height = 14,
#        units = c("cm"))
# 
# }}
```

```{r}
# 
# donors_loop<-c("D1","D2", "D3", "D4", "D5", "D6", "D7", "D8")
# stress_level_cond<-c("control", "low", "median", "high", "max")
# 
# 
# phyloseq_rare%>%
#   subset_samples(time %in% c("feces", "pre-stress" , "stress", "post_stress_1", "post_stress_2" ))->phyloseq_bar
# 
# phyloseq_bar%>%
#   sample.data.frame()->meta
# 
# 
# 
# meta%>%
#   mutate(stress_level = ifelse(time == "pre-stress", "pre", stress_level))%>%
#   mutate(time= factor(time, levels=c("feces", "pre-stress", "stress", "post_stress_1", "post_stress_2")))->meta
#                                             
# #update the metadata file
# sample_data(phyloseq_bar)<-meta
# 
# 
# 
# for (i in 1:length(donors_loop)){
#   for (j in 1:length(stress_level_cond)){
# 
# phyloseq_bar%>% 
#     subset_samples(donor_name %in% c(donors_loop[i]))%>%
#      subset_samples(stress %in% c("O2","control", "feces"))%>%
#       subset_samples(stress_level %in% c(stress_level_cond[j],"pre", "feces"))%>%
#       subset_samples(incubation %in% "anaerobe" | (stress_level %in% c("pre", "feces")))%>%
#      ps_arrange(time)%>%
#       microViz::comp_barplot(
#     tax_level = "Family",
#     label = "time",
#     tax_order = sum, 
#     sample_order = "default",
#     n_taxa = 10,
#     bar_width = 0.9,
#     tax_transform_for_plot = "compositional",
#       palette= myPal) +
#     ylab("Proportion") + 
#     xlab( "")-> p_hist
# 
#  
# p_hist + 
#   # facet_grid(cols=vars(stress_level), drop = TRUE, scales = "free_x", space="free")+
#   theme(axis.text.x = element_text(size=20, angle=90, vjust=0.2, hjust=1, margin=margin(-8,0,0,0)),
#         axis.ticks.x = element_blank(),strip.text = element_text(size  = 20),
#         axis.text.y= element_text(size = 20), 
#         axis.title.y = element_text(size=20), 
#         axis.title.x = element_text(size=20),
#         legend.text = element_text(size = 15, vjust=1, hjust=0),
#         legend.title = element_text(size= 20),
#         plot.title = element_text(size=20), panel.spacing = unit(0.2, "cm")) + 
#   scale_y_continuous(breaks=c(0.25, 0.5, 0.75))+ 
#   ggtitle(paste(donors_loop[i], "-", stress_level_cond[j],  " stress")) -> p_hist
# 
# p_hist
# 
# 
# 
# x<-paste0(donors_loop[i],stress_level_cond[j],"H2O2_stress_box.jpeg")
# ggsave(filename = x,
#        path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/O2_bar/anaerobe",
#        width = 14,
#        height = 14,
#        units = c("cm"))
# 
# }}
```



##line plots 
###On Genus level: pure culture data: H2O2
```{r}
tax_level ="Genus"

ps_strain_filt%>%
  subset_samples(!donor_name %in% c("D1"))%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()->df_abund 
  
 
calculate_q1 <- function(x) {
  return(quantile(x, 0.25))
}
calculate_q3 <- function(x) {
  return(quantile(x, 0.75))
}

left_join(left_join(setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q1), c("time", "stress_level", "stress",tax_level,"incubation", "q3" )), 
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q3), c("time", "stress_level", "stress",tax_level,"incubation", "q1" ))),
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=median), c("time", "stress_level", "stress", tax_level,"incubation", "abund")))->means


# add the pre-stress cond for all 
#####
means %>%
  subset(!time %in% c("pre-stress", "feces"))->no
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control")->ctrl
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low")->low
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median")->median
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high")->high
combined_df <- do.call(rbind, list(no, ctrl, low, median, high))
######

# add control for all stress level
###

combined_df%>%
  subset(!stress_level %in% c("control"))%>%
   mutate(incubation = "stress")->no
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "low", incubation = "control")->low
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "median", incubation = "control")->median
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "high", incubation = "control")->high
combined_df_control <- do.call(rbind, list(no, low, median, high))

```

```{r}
filter_decreased<-c("Bacteroides","Fusicatenibacter","Anaerostipes","Dorea","Blautia","Agathobacter","Faecalibacterium","Lachnoclostridium","Lachnospiraceae UCG-004","Lachnospiraceae Family" , "[Eubacterium] hallii group" ,"Parabacteroides",   "Lachnospiraceae ND3007 group")


combined_df_control%>%
  subset(stress %in% c("H2O2", "control"))%>%
  subset(stress_level %in% c("median", "high") )%>%
  # filter(abund > 0 )%>%
  mutate(stress_level = factor(stress_level, levels=c(  "median", "high")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
  subset(Genus %in% filter_decreased)->sub

setNames(aggregate(sub$abund, by=list(sub$Genus, sub$stress, sub$time), FUN=mean), c("Genus", "stress", "time","mean_abund"))%>%
  subset(stress %in% "control" & time %in% "pre-stress")->order

left_join(combined_df_control %>% subset(Genus %in% filter_decreased), order[,c("Genus", "mean_abund")])%>%
  subset(stress %in% c("H2O2", "control"))%>%
  # filter(abund > 0 )%>%
  subset(stress_level %in% c("median", "high") )%>%
  mutate(stress_level = factor(stress_level, levels=c(  "median", "high")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
  # subset(Genus %in% filter_increased)%>%
  mutate(Genus = reorder(Genus, abund, decreasing = T))%>%
  ggplot(aes(x=time, y=abund))+
  geom_line(aes(colour=incubation, group=incubation))+
  geom_ribbon(aes(ymin = q1, ymax = q3, fill=incubation, group=incubation),
              alpha = 0.3)+
  geom_vline(aes(xintercept = which(levels(time) == 'stress')),  # Specify your desired category label
             linetype = "dashed", color = "coral4", alpha=0.8, size=0.5) +
   geom_vline(aes(xintercept = which(levels(time) == 'post_stress_1')),  # Specify your desired category label
             color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_2')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'pre-stress')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  facet_grid(cols=vars(stress_level), rows = vars(Genus), scale="free")+
  theme(text = element_text(size=15),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_text(size=15),
        axis.text.y=element_text(size=15),
        strip.text.y = element_text(angle=0, hjust=0, face="italic"),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text = element_text(size=15),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))+
  ylab("Clr-abundance")+
   scale_fill_manual(values=c( "skyblue3", "lightpink3"), guide="none")+
  scale_colour_manual(values=c("skyblue3", "lightpink3"), labels=c("Control", "Stress"), name="Condition")+
  scale_x_discrete(labels=c('pre-stress' = "Pre", 'stress' = "Stress", 'post_stress_1' = "Post 1", 'post_stress_2' = "Post 2" ))+
  scale_y_continuous(breaks = c(0, 2.5, 5), expand = c(0, 0)) +
  geom_hline(yintercept=0, lty=2, colour="grey")


ggsave(filename = paste( "pure_culture_H2O2_decreased.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 20,
       height = 28,
       units = c("cm"))

#decrease
#increased format 20x28
146/25*18
146/25*13

14*2.26
```


```{r}
filter_increased<-c("Escherichia-Shigella","Enterococcus","Bifidobacterium","Lactobacillales Order")

combined_df_control%>%
  subset(stress %in% c("H2O2", "control"))%>%
  subset(stress_level %in% c("median", "high") )%>%
  # filter(abund > 0 )%>%
  mutate(stress_level = factor(stress_level, levels=c(  "median", "high")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
  subset(Genus %in% filter_increased)->sub

setNames(aggregate(sub$abund, by=list(sub$Genus, sub$stress, sub$time), FUN=mean), c("Genus", "stress", "time","mean_abund"))%>%
  subset(stress %in% "control" & time %in% "pre-stress")->order

left_join(combined_df_control %>% subset(Genus %in% filter_increased), order[,c("Genus", "mean_abund")])%>%
  subset(stress %in% c("H2O2", "control"))%>%
  # filter(abund > 0 )%>%
  subset(stress_level %in% c("median", "high") )%>%
  mutate(stress_level = factor(stress_level, levels=c(  "median", "high")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
  # subset(Genus %in% filter_increased)%>%
  mutate(Genus = reorder(Genus, abund, decreasing = T))%>%
  ggplot(aes(x=time, y=abund))+
  geom_line(aes(colour=incubation, group=incubation))+
  geom_ribbon(aes(ymin = q1, ymax = q3, fill=incubation, group=incubation),
              alpha = 0.3)+
  geom_vline(aes(xintercept = which(levels(time) == 'stress')),  # Specify your desired category label
             linetype = "dashed", color = "coral4", alpha=0.8, size=0.5) +
   geom_vline(aes(xintercept = which(levels(time) == 'post_stress_1')),  # Specify your desired category label
             color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_2')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'pre-stress')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  facet_grid(cols=vars(stress_level), rows = vars(Genus), scale="free")+
  theme(text = element_text(size=15),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_text(size=15),
        axis.text.y=element_text(size=15),
        strip.text.y = element_text(angle=0, hjust=0, face="italic"),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text = element_text(size=15),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))+
  ylab("Clr-abundance")+
   scale_fill_manual(values=c( "skyblue3", "lightpink3"), guide="none")+
  scale_colour_manual(values=c("skyblue3", "lightpink3"), labels=c("Control", "Stress"), name="Condition")+
  scale_x_discrete(labels=c('pre-stress' = "Pre", 'stress' = "Stress", 'post_stress_1' = "Post 1", 'post_stress_2' = "Post 2" ))+
  scale_y_continuous(breaks = c(0, 2.5, 5), expand = c(0, 0)) +
  geom_hline(yintercept=0, lty=2, colour="grey")


ggsave(filename = paste( "pure_culture_H2O2_increased.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 18,
       height = 11.5,
       units = c("cm"))

#decrease
#increased format 20x28
146/25*18
146/25*11.5

14*2.26
```


#on Genus level- O2

```{r}
tax_level ="Genus"


ps_strain_filt%>%
   # subset_samples(!donor_name %in% c("D1", "D5"))%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()->df_abund 
  
  
  
# left_join(setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=mean), c("time", "stress_level", "stress", tax_level,"incubation", "abund")),
#             setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=sd), c("time", "stress_level", "stress",tax_level,"incubation", "sd")))->means


calculate_q1 <- function(x) {
  return(quantile(x, 0.25))
}
calculate_q3 <- function(x) {
  return(quantile(x, 0.75))
}

left_join(left_join(setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q1), c("time", "stress_level", "stress",tax_level,"incubation", "q3" )), 
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q3), c("time", "stress_level", "stress",tax_level,"incubation", "q1" ))),
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=median), c("time", "stress_level", "stress", tax_level,"incubation", "abund")))->means


means %>%
  subset(!time %in% c("pre-stress", "feces"))->no

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control")->ctrl

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low")->low

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median")->median
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high")->high
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "max")->max

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control", incubation = "aerobe")->ctrl_a

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low", incubation = "aerobe")->low_a

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median", incubation = "aerobe")->median_a
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high", incubation = "aerobe")->high_a
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "max", incubation = "aerobe")->max_a

combined_df <- do.call(rbind, list(no, ctrl, low, median, high, max, ctrl_a, low_a, median_a, high_a, max_a))

# combined_phylum<-left_join(combined_df, as.data.frame(tax_table(ps_strain_filt))[, c("Phylum", "Family")])


# setNames(aggregate(combined_phylum$abund, by=list(combined_phylum$Family), FUN=mean), c("Family", "abund"))%>%
#   subset(abund > 1)->filter
# filter <- c("Escherichia-Shigella", "Bacteroides")
filter <- c("Fusicatenibacter")

filter_2<- c("Faecalibacterium", "Roseburia", "Eubacterium", "Anaerostipes", "Bifidobacterium", "Parabacteroides", "Agathobacter", "Fusicatenibacter", "Anaerogotignum", "Clostridium", "Colinsella", "Enterococcus", "Lactiplantibacillus", "Phascolarctobacterium", "Prevotella", "Ruminococcus")

combined_df%>%
  subset(stress %in% c("O2", "control"))%>%
  subset(Genus %in% c(filter))%>%
 # filter(abund > 0 )%>%
   mutate(stress_level = ifelse(stress_level == "control", "minimal", stress_level))%>%
  mutate(stress_level = factor(stress_level, levels=c("minimal", "low", "median", "high", "max")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
   mutate(Genus = reorder(Genus, abund, decreasing = T))%>%
  ggplot(aes(x=time, y=abund))+
  # geom_hline(yintercept=0)+
  geom_line(aes(colour=incubation, group=incubation))+
  geom_ribbon(aes(ymin =q3, ymax = q1, fill=incubation, group=incubation),
              alpha = 0.3)+
  facet_grid(cols=vars(stress_level), rows = vars(Genus), scale="free")+
  theme(text = element_text(size=15), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_text(size=15),
        axis.text.y=element_text(size=12),
        strip.text.y = element_text(angle=0, face="italic", hjust=0),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text.x = element_text(size=15, hjust=0.5),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))+
  scale_fill_manual(values=c( "lightpink3","skyblue3"), guide="none")+
  scale_colour_manual(values=c("lightpink3","skyblue3"), labels=c("Stress", "Control"))+
  ylab("Clr-abundance") +
  geom_vline(aes(xintercept = which(levels(time) == 'stress')),  # Specify your desired category label
             linetype = "dashed", color = "coral4", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_1')),  # Specify your desired category label
             color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_2')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'pre-stress')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  scale_x_discrete(labels=c('pre-stress' = "Pre", 'stress' = "Stress", 'post_stress_1' = "Post 1", 'post_stress_2' = "Post 2" ))+
  labs(colour="Condition")+
  ylab("Clr-difference")

ggsave(filename = paste( "increase_O2_box.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 21,
       height = 7,
       units = c("cm"))


146/25*21
146/25*7
```

##donor specififc increase of taxa with h2o2
```{r}
# install.packages("ggdendro")
# # install.packages("grid")
# library(ggdendro)
# library(grid)
# remotes::install_github("wilkelab/cowplot")
# library(cowplot)

tax_level ="Genus"


ps_strain_filt%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()%>%
  subset(stress %in% c("H2O2"))->df_abund 

# calculate the difference
df_abund[,c("Abundance", "stress_level", "time", "Genus",  "donor_name")]%>%
  pivot_wider(names_from=stress_level, values_from = Abundance)%>%
  mutate(diff_high = high- control,
         diff_median = median - control,
         diff_low= low - control)-> clr_diff_h2o2

##create a filter dataframe based on min. abundance of 0
clr_diff_h2o2 %>%
  subset(time %in% "stress")%>%
  subset(control > 1 | low > 1 | median > 1 | high > 1)%>%
  mutate(cut_off = "yes")-> df_cutoff

left_join(clr_diff_h2o2, df_cutoff[, c("donor_name", "cut_off", "Genus")] )%>%
  subset(diff_high > 1 | diff_high < -1 )%>%
  subset(time %in% "stress")->filter

filter_increased<-unique(filter$Genus)
  

clr_diff_h2o2%>%
  subset(Genus %in% filter_increased) %>%  subset(time %in% "stress")->dendro


dendro[,c("Genus", "donor_name", "diff_high")]%>%
  pivot_wider(values_from = diff_high, names_from = Genus)->df_diff

matrix_diff<-as.matrix(df_diff[,-c(1)])
rownames(matrix_diff)<-df_diff$donor_name
  
diff_dendro <- as.dendrogram(hclust(d = dist(x = matrix_diff)))
# Create dendro
dendro_plot <- ggdendrogram(data = diff_dendro, rotate = F)
diff_order <- order.dendrogram(diff_dendro)



left_join(clr_diff_h2o2, df_cutoff[, c("donor_name", "cut_off", "Genus")] )%>%
  subset( high > 1 | median > 1 |low > 1 | control > 1) %>%
  mutate(time = factor(time, levels=c("stress", "post_stress_1", "post_stress_2")))%>%
  subset(Genus %in% filter_increased)%>%
  gather(condition, diff, "diff_high":"diff_median")%>%
  mutate(donor_name =  factor(donor_name, levels = df_diff$donor_name[diff_order], ordered = TRUE))%>%
  mutate(Genus =  reorder(Genus, diff, decreasing = T))%>%
  ggplot(aes(y=condition, x= time, fill =diff))+
  geom_tile() +
  facet_grid(rows=vars(Genus), cols=vars(donor_name), scales="free")+
  scale_fill_gradientn(colors = c("coral", "white", "cyan4"), na.value= "grey",values = rescale(c(min(dendro$diff_high),0,max(dendro$diff_high))))+ theme(text = element_text(size=15),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_blank(),
        axis.text.y=element_text(size=12),
        strip.text.y = element_text(angle=0, hjust=0, face="italic"),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text = element_text(size=15),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))-> heatmap_plot
  

  
plot_grid(dendro_plot, heatmap_plot, 
          rel_heights = c(0.4, 1), 
          nrow=2,
          ncol=1,
          scale=c(0.5,1))


ggsave(filename = paste( "increase_H2O2_box_median.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 40,
       height = 80,
       units = c("cm"))
  
```


##donor specififc increase of taxa with o2
```{r}
# install.packages("ggdendro")
# # install.packages("grid")
# library(ggdendro)
# library(grid)
# remotes::install_github("wilkelab/cowplot")
# library(cowplot)

tax_level ="Genus"


ps_strain_filt%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()%>%
  subset(stress %in% c("O2"))->df_abund 
##calculate the difference 
df_abund[,c("Abundance", "stress_level", "time", "Genus",  "donor_name", "incubation")]%>%
  pivot_wider(names_from=incubation, values_from = Abundance)%>%
  mutate(diff = aerobe - anaerobe)-> clr_diff_o2

##create a filter dataframe based on min. abundance of 0
clr_diff_o2 %>%
  subset(time %in% "stress" & stress_level %in% "max")%>%
  subset(aerobe > 1 | anaerobe > 1)%>%
  mutate(cut_off = "yes")-> df_cutoff
  
##create a filter dataframe based on clr increase
left_join(clr_diff_o2, df_cutoff[, c("donor_name", "cut_off", "Genus")] )%>%
  subset(cut_off %in% "yes") %>%
  subset(diff > 1.5 | diff < -1.5 | Genus %in% "Bacteroides")%>%
   # subset(diff < - 1 )%>%
  subset(time %in% "stress")->filter

filter<-unique(filter$Genus)
  

left_join(clr_diff_o2, df_cutoff[, c("donor_name", "cut_off", "Genus")] )%>%
  subset(Genus %in% filter) %>%  
  subset(cut_off %in% "yes") %>%
  subset(time %in% "stress" & stress_level %in% "max")->dendro


dendro[,c("Genus", "donor_name", "diff")]%>%
  pivot_wider(values_from = diff, names_from = Genus)->df_diff

matrix_diff<-as.matrix(df_diff[,-c(1)])
rownames(matrix_diff)<-df_diff$donor_name
  
diff_dendro <- as.dendrogram(hclust(d = dist(x = matrix_diff)))
# Create dendro
dendro_plot <- ggdendrogram(data = diff_dendro, rotate = F)
diff_order <- order.dendrogram(diff_dendro)



left_join(clr_diff_o2, df_cutoff[, c("donor_name", "cut_off", "Genus")] )%>%
  subset(cut_off %in% "yes") %>%
  mutate(time = factor(time, levels=c("stress", "post_stress_1", "post_stress_2")))%>%
  subset(Genus %in% filter )%>%
  mutate(donor_name =  factor(donor_name, levels = df_diff$donor_name[diff_order], ordered = TRUE),
         stress_level = factor(stress_level, levels = c("control", "low", "median", "high", "max")))%>%
   # mutate(donor_name =  factor(donor_name, levels = df_diff$donor_name[diff_order], ordered = TRUE))%>%
  mutate(Genus =  reorder(Genus, diff, decreasing = T))->sub

sub%>%
  ggplot(aes(y=stress_level, x= time, fill =diff))+
  geom_tile() +
  facet_grid(rows=vars(Genus), cols=vars(donor_name), scales="free")+
  scale_fill_gradientn(colors = c("coral", "white", "cyan4"), na.value= "grey",values = rescale(c(min(sub$diff),0,max(sub$diff))))+ theme(text = element_text(size=15),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_blank(),
        axis.text.y=element_text(size=12),
        strip.text.y = element_text(angle=0, hjust=0, face="italic"),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text = element_text(size=15),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))-> heatmap_plot
  

  
plot_grid(dendro_plot, heatmap_plot, 
          rel_heights = c(0.4, 1), 
          nrow=2,
          ncol=1,
          scale=c(0.5,1))


ggsave(filename = paste( "increase_O2_box_median.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 40,
       height = 80,
       units = c("cm"))
  
```

## h2o2 taxa specififc clr abund - for later !!;P
```{r}
tax_level ="Genus"

ps_strain_filt%>%
  subset_samples(!donor_name %in% c("D1"))%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()->df_abund 
  
 
calculate_q1 <- function(x) {
  return(quantile(x, 0.25))
}
calculate_q3 <- function(x) {
  return(quantile(x, 0.75))
}

left_join(left_join(setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q1), c("time", "stress_level", "stress",tax_level,"incubation", "q3" )), 
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=calculate_q3), c("time", "stress_level", "stress",tax_level,"incubation", "q1" ))),
setNames(aggregate(df_abund$Abundance, by=list(df_abund$time, df_abund$stress_level, df_abund$stress, df_abund[[tax_level]], df_abund$incubation), FUN=median), c("time", "stress_level", "stress", tax_level,"incubation", "abund")))->means


# add the pre-stress cond for all 
#####
means %>%
  subset(!time %in% c("pre-stress", "feces"))->no
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control")->ctrl
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low")->low
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median")->median
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high")->high
combined_df <- do.call(rbind, list(no, ctrl, low, median, high))
######

# add control for all stress level
###

combined_df%>%
  subset(!stress_level %in% c("control"))%>%
   mutate(incubation = "stress")->no
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "low", incubation = "control")->low
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "median", incubation = "control")->median
combined_df %>%
  subset(stress_level %in% "control")%>%
  mutate(stress_level = "high", incubation = "control")->high
combined_df_control <- do.call(rbind, list(no, low, median, high))

```


## o2 taxa specififc clr abund - for later !!;P
```{r}
tax_level ="Genus"


ps_strain_filt%>%
  physeq_glom_rename(taxrank = tax_level, 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt()->means 


means %>%
  subset(!time %in% c("pre-stress", "feces"))->no

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control")->ctrl

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low")->low

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median")->median
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high")->high
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "max")->max

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "control", incubation = "aerobe")->ctrl_a

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "low", incubation = "aerobe")->low_a

means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "median", incubation = "aerobe")->median_a
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "high", incubation = "aerobe")->high_a
means %>%
  subset(time %in% "pre-stress")%>%
  mutate(stress_level = "max", incubation = "aerobe")->max_a

combined_df <- do.call(rbind, list(no, ctrl, low, median, high, max, ctrl_a, low_a, median_a, high_a, max_a))


filter <- c("Sutterella")


combined_df%>%
  subset(stress %in% c("O2", "control"))%>%
  subset(Genus %in% c(filter))%>%
 # filter(abund > 0 )%>%
  mutate(stress_level = ifelse(stress_level == "control", "minimal", stress_level))%>%
   mutate(stress_level = factor(stress_level, levels=c("minimal", "low", "median", "high", "max")))%>%
  mutate(time = factor(time, levels=c("pre-stress", "stress", "post_stress_1", "post_stress_2")))%>%
  #  mutate(Genus = reorder(Genus, abund, decreasing = T))%>%
  ggplot(aes(x=time, y=Abundance))+
  geom_line(aes(colour=incubation, group=incubation))+
  facet_grid(cols=vars(stress_level), rows = vars(donor_name), scale="free_x")+
  theme(text = element_text(size=15), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15, angle = 90, vjust= 0.5, hjust=1),
        axis.title.y= element_text(size=15),
        axis.text.y=element_text(size=12),
        strip.text.y = element_text(angle=0, face="italic", hjust=0),
        strip.background = element_blank(),
        axis.ticks.y = element_blank(),
        legend.text = element_text(size = 15),
        legend.title = element_text(size= 15),
        plot.title = element_text(size=20, hjust=0.25),
        strip.text.x = element_text(size=15, hjust=0.5),
        panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))+
  scale_fill_manual(values=c( "lightpink3","skyblue3"), guide="none")+
  scale_colour_manual(values=c("lightpink3","skyblue3"), labels=c("Stress", "Control"))+
  ylab("Clr-abundance") +
  geom_vline(aes(xintercept = which(levels(time) == 'stress')),  # Specify your desired category label
             linetype = "dashed", color = "coral4", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_1')),  # Specify your desired category label
             color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'post_stress_2')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  geom_vline(aes(xintercept = which(levels(time) == 'pre-stress')),  # Specify your desired category label
              color = "grey", alpha=0.8, size=0.5) +
  scale_x_discrete(labels=c('pre-stress' = "Pre", 'stress' = "Stress", 'post_stress_1' = "Post 1", 'post_stress_2' = "Post 2" ))+
  labs(colour="Condition")+
  ylab("Clr-difference")

ggsave(filename = paste( "increase_O2_box.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output",
       width = 21,
       height = 7,
       units = c("cm"))


146/25*21
146/25*7
```
