
```{r setup, echo =FALSE}
.libPaths('C:\\Users\\zuendj\\Documents\\R4Libs')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "P:/Shared_documents/Serafina PlÃ¼ss/20230320_sequencing_results/Janina", cho = TRUE, warning = FALSE, message = FALSE)

# library("plyr")
# install.packages("vctrs")
# install.packages("vegan")
# install.packages("cli")

# library(vctrs)
# library(vegan)
# install.packages("ggplot2")
#library(ggplot2)
#install.packages('tinytex')
#install.packages('rmarkdown')
#install.packages("rlang")
#install.packages("nlme")
#install.packages("mgcv")
#install.packages("ape")
#library(nlme)
# install.packages("remotes")
# remotes::install_github("mikemc/speedyseq")
# install.packages("BiocManager")
# BiocManager::install("DESeq2")
#install.packages("fs")
# install.packages('devtools')
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
#unload("mia")
# install.packages("GUniFrac")

# install.packages(
#   "microViz",
#   repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos"))
# )
# devtools::install_github("tidyverse/tidyverse")
 # library(tidyverse)
  # library(openxlsx)
# # install.packages("ggalluvial")
# library(ggalluvial)
 # library(ggh4x)
# library(GUniFrac)
 # library(rstatix)
# library(tinytex)
# library(rmarkdown)
# library(rlang) 
# library(tidyverse)
# library(microViz)
# library(dplyr)
# library(mgcv)
# library(ape)
# library(phyloseq)
# library(speedyseq)
# library(DESeq2)
 # library(plyr)
# library(fs)
# library(devtools)
# library(pairwiseAdonis)
# library(ggpubr)

# install.packages("dplyr")
# install.packages("data.table")
# devtools::install_github("vmikk/metagMisc")
#  
# library(metagMisc)


```

```{r, echo =FALSE}
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_normalisation.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_alpha.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R") 
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_heatmap.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_taxa_tests.R")
source("https://raw.githubusercontent.com/fconstancias/metabaRpipe-source/master/Rscripts/functions.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_varia.R")
source("https://raw.githubusercontent.com/fconstancias/DivComAnalyses/master/R/phyloseq_beta.R")

#get metadata file: load this function
sample.data.frame <- function(ps) {
  return(as(phyloseq::sample_data(ps), "data.frame"))}
```

## 1. Get the phyloseq object 
```{r}
# unloadNamespace("miaViz")
# unloadNamespace("mia")

phyloseq <-readRDS("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/phyloseq.RDS")
phyloseq%>%
  phyloseq_get_strains()->phyloseq
```


```{r}
hplc<-read.xlsx("C:/Users/zuendj/Desktop/03_data R/wp2_complex/input/20230613_hplc_list_results_stress.xlsx")
hplc%>%
  subset(!sample_name %in% NA)->hplc

seq<-read.xlsx("P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/mapping_file.xlsx")

meta<-left_join(seq, hplc[,c(1:12)])
write.xlsx(meta, "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta_new.xlsx")
```

### Get and update metadata file
```{r, echo=FALSE}

phyloseq%>%
  physeq_add_metadata(physeq = .,
                      metadata = "P:/Shared_documents/Janina_Zuend/08_phd/01_sequencing_data/Ox_stress/meta.xlsx" %>%
                        readxl::read_xlsx(),
                      sample_column = "sample_name") -> phyloseq

as.data.frame(sample_data(phyloseq))->meta
## remove donor 2 from data as very low sample size
# phyloseq %>%
#   subset_samples(!donor_name %in% "D1")->phyloseq



```

```{r}
ps_strain_filt <- phyloseq

# threshold in %
threshold = 0.25

# filter per sample to remove all ASV lower in abundance than threshold (%)
otu_table(ps_strain_filt) <- otu_table(ps_strain_filt) %>%
  as.data.frame() %>%
  dplyr:: mutate(across(everything(), ~ ifelse(. <= sum(.) * (threshold/100), 0, .))) %>% otu_table(., taxa_are_rows = TRUE)

# remove all ASVs that are not present in any sample
ps_strain_filt <- ps_strain_filt %>%
  filter_taxa(function(x) sum(x > 0) > 0, TRUE)
```



#Differtial abunance
```{r, echo=FALSE}
#BiocManager::install("microbiome/mia")

 library(mia)
# library(miaViz)
 library(ALDEx2)
```

```{r}
ox <- makeTreeSummarizedExperimentFromPhyloseq(ps_strain_filt) 
```


```{r}
ox_fam <- ox %>% agglomerateByRank(rank = "Family")
ox_gen <- ox %>% agglomerateByRank(rank = "Genus")
ox_species <- ox %>% agglomerateByRank(rank = "Species") 
```



###Aldex: NO Prevalence filter-> filter later with abundance

```{r}
oxygen <-ox_gen[ , ox_gen$stress %in% c("O2") ]
stress_level_cond <-c("control", "low","median", "high", "max")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_oxygen_genus <-data.frame(matrix(ncol=15, nrow=0))

for (i in (1:length(stress_level_cond))){
  ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) & !oxygen$donor_name %in% c("D1")]
  
  # ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) & !oxygen$donor_name %in% c("D1")]
  
  
    for (j in (1:length(time_cond))){
  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
    result_name <- paste(stress_level_cond[i])
   
      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$incubation)

      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- paste(time_cond[j])
      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_oxygen_genus <- rbind(all_oxygen_genus, result)
      }}
all_oxygen_genus %>%
  filter(we.ep < 0.05)->out

out[,c(1,8,10, 14,15)]
```

```{r}
oxygen <-ox_species[ , ox_species$stress %in% c("O2") ]
stress_level_cond <-c("control", "low","median", "high", "max")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_oxygen_species <-data.frame(matrix(ncol=15, nrow=0))

for (i in (1:length(stress_level_cond))){
  ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i]) & !oxygen$donor_name %in% c("D1")]
    for (j in (1:length(time_cond))){
  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
    result_name <- paste(stress_level_cond[i])
   
      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$incubation)

      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- paste(time_cond[j])
      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_oxygen_species <- rbind(all_oxygen_species, result)
      }}
all_oxygen_species %>%
  filter(we.ep < 0.05)->out

out[,c(1,8,10, 14,15)]
```

```{r}
# oxygen <-ox_fam[ , ox_fam$stress %in% c("O2") ]
# stress_level_cond <-c("control", "low","median", "high", "max")
# time_cond<-c("stress", "post_stress_1", "post_stress_2")
# all_oxygen <-data.frame(matrix(ncol=15, nrow=0))
# 
# for (i in (1:length(stress_level_cond))){
#   ox_sub <-oxygen[ , oxygen$stress_level %in% c(stress_level_cond[i])]
#     for (j in (1:length(time_cond))){
#   ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
#     result_name <- paste(stress_level_cond[i])
#    
#       x <- aldex.clr(
#       reads = assay(ox_sub_sub),
#      ox_sub_sub$incubation)
# 
#       x_tt <- aldex.ttest(
#         x, 
#         paired.test = FALSE, 
#         verbose = FALSE)
#       
#       x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
#       aldex_out <- data.frame(x_tt, x_effect)
#       
#       result <-aldex_out
#       result[,14] <- result_name
#       result[,15] <- paste(time_cond[j])
#       names(result)[6]<-paste0("rab.win.aero")
#       names(result)[7]<-paste0("rab.win.anaero")
#       result$Family<-row.names(result)
#       all_oxygen <- rbind(all_oxygen, result)
#       }}
# all_oxygen %>%
#   filter(effect < -1 | effect > 1)->out
# out[,c(1,8,10, 14,15)] %>%
#   subset (V15 %in% "stress")
```

###Aldex: H2O2
###species level
```{r}
h2o2 <-ox_species[ , ox_species$stress %in% c("H2O2")]

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_h2o2_species <-data.frame(matrix(ncol=15, nrow=0))
j=0 

for (i in (1:length(stress_level_cond))){
  ox_sub <-h2o2[ , h2o2$stress_level %in% c(stress_level_cond[i], "control")]

  for (j in (1:length(time_cond))){

  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
  
     ox_sub_sub <- subsetByPrevalentTaxa(ox_sub_sub, detection = 0, prevalence=0.25, include_lowest= TRUE)
      result_name <- paste(stress_level_cond[i])

      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$stress_level)
      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- time_cond[j]

      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_h2o2_species <- rbind(all_h2o2_species, result)
        }}
```

```{r}
h2o2 <-ox_gen[ , ox_gen$stress %in% c("H2O2")]

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_h2o2_genus <-data.frame(matrix(ncol=15, nrow=0))
j=0 

for (i in (1:length(stress_level_cond))){
  ox_sub <-h2o2[ , h2o2$stress_level %in% c(stress_level_cond[i], "control") & !(h2o2$donor_name %in% "D1")]

  for (j in (1:length(time_cond))){

  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
  
      ox_sub_sub <- subsetByPrevalentTaxa(ox_sub_sub, detection = 0, prevalence=0.5, include_lowest= TRUE)
      result_name <- paste(stress_level_cond[i])

      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$stress_level)
      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- time_cond[j]

      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_h2o2_genus <- rbind(all_h2o2_genus, result)
        }}
```
####Family level
```{r}
h2o2 <-ox_fam[ , ox_fam$stress %in% c("H2O2")]
stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
all_h2o2 <-data.frame(matrix(ncol=15, nrow=0))

for (i in (1:length(stress_level_cond))){
  ox_sub <-h2o2[ , h2o2$stress_level %in% c(stress_level_cond[i], "control")]
  
  for (j in (1:length(time_cond))){
  ox_sub_sub <-ox_sub[ , ox_sub$time %in% c(time_cond[j])]
      ox_sub_sub <- subsetByPrevalentTaxa(ox_sub_sub, detection = 0, prevalence=0.5, include_lowest= TRUE)
      result_name <- paste(stress_level_cond[i])

      x <- aldex.clr(
      reads = assay(ox_sub_sub),
     ox_sub_sub$stress_level)
      x_tt <- aldex.ttest(
        x, 
        paired.test = FALSE, 
        verbose = FALSE)
      x_effect <- aldex.effect(x, CI = TRUE, verbose = FALSE)
      aldex_out <- data.frame(x_tt, x_effect)
      
      result <-aldex_out
      result[,14] <- result_name
      result[,15] <- paste(time_cond[j])

      names(result)[6]<-paste0("rab.win.aero")
      names(result)[7]<-paste0("rab.win.anaero")
      result$Family<-row.names(result)
      all_h2o2 <- rbind(all_h2o2, result)
        }}

all_h2o2 %>%
  filter(wi.eBH < 0.05)->out
out[,c(1,8,10, 14,15, 16)]

unloadNamespace("miaViz")
unloadNamespace("mia")
```
###CLR differences
#O2
```{r}
phyloseq%>%
  subset_samples(stress %in% "O2")%>%
  physeq_glom_rename(taxrank = "Family",
                     speedyseq = TRUE)%>%
   microbiome::transform("clr") %>%
   psmelt() ->df_abund

clr_O2<-left_join(df_abund, as.data.frame(phyloseq@tax_table))

mean_abund<-setNames(aggregate(clr_O2$Abundance, by=list(clr_O2$Family), FUN = "mean"), c("Family", "abund"))

clr_O2<-left_join(clr_O2, mean_abund)

mean_abund$Family<-as.factor(mean_abund$Family)

clr_O2%>%
  mutate(Family = fct_reorder(Family, clr_O2$abund)) ->hoi

ordered<-unique(as.character(hoi$Family))[1:15]
# clr_drug$treatment<-factor(clr_drug$treatment, levels=c("DMSO", "omeprazole",  "ciprofloxacin","b_5-FU","a_control"))

clr_O2%>%
  subset(Family %in% c(ordered))->clr_O2

clr_diff<-setNames(as.data.frame(matrix(0,0,5)), c("diff", "Family", "stress_level", "donor_name", "time"))

stress_level_cond <-c("control", "low","median", "high", "max")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_O2$donor_name)
donor_fam<-unique(clr_O2$Family)


i=1
for (q in 1: length(time_cond)){
  clr_O2%>%
    subset(time %in% time_cond[q])->sub1a

for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% stress_level_cond[m])->sub1

  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2

    for (k in 1:length(donor_fam)){
    sub2%>%
    subset(Family %in% donor_fam[k])->sub3

      if (nrow(sub3) > 0){

      sub3 %>%
          subset(incubation %in% "aerobe")->aerobe
       sub3 %>%
          subset(incubation %in% "anaerobe")->anaerobe

      clr_diff[i, c("diff")] <- aerobe$Abundance - anaerobe$Abundance
      clr_diff[i, c("Family")] <-donor_fam[k]
      clr_diff[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff[i, c("donor_name")] <-donor_cond[j]
      clr_diff[i, c("time")] <-time_cond[q]
  i=i+1}
    }}}}

clr_diff<-left_join(clr_diff, mean_abund)
stats_o2<-setNames(all_oxygen[, c("we.ep", "V14", "V15", "Family")], c("p", "stress_level", "time", "Family"))
stats_o2$Family<-str_remove(stats_o2$Family, "Family:")

for (i in 1:nrow(stats_o2)){
  if (stats_o2[i,c("p")] <0.05 & stats_o2[i,]$p >0.01){stats_o2[i,c("sig")] = "*"}
  if (stats_o2[i,c("p")] <0.01 & stats_o2[i,]$p >0.001){stats_o2[i,c("sig")] = "**"}
  if (stats_o2[i,c("p")] <0.001 ){stats_o2[i,c("sig")] = "***"}
}

diff_stats_o2<-join(clr_diff, stats_o2)
diff_stats_o2$stress_level<-factor(diff_stats_o2$stress_level, levels=c("control", "low","median", "high", "max"))
diff_stats_o2$time<-factor(diff_stats_o2$time, levels=c("stress", "post_stress_1","post_stress_2"))

diff_stats_o2%>%
  mutate(Family = fct_reorder(Family, diff_stats_o2$abund))%>%
  ggplot(aes(x=diff, y=Family))+
geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
   geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_jitter(aes(group=stress_level, colour=donor_name), position = position_jitterdodge(0.1) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"),
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"),
              axis.title.y = element_text(size=20),
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20),
         strip.text = element_text(size=20),
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA))+
      ylab("")+xlab("Differences in clr-abundance")+labs(colour="Donor")+
  facet_grid(cols=vars(time))+
  geom_text(data= subset(diff_stats_o2, donor_name %in% c("D1")), aes(label=sig, group=stress_level), position = position_dodge(width=0.75), size=5, colour="red")
      ggtitle("O2-specifically altered families")
ggsave(filename = paste0("clr_dist.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/O2/clr_diff",
       width = 25,
       height = 30,
       units = c("cm"))

51.7/14*27
65.3/17*30
```

## Genus level
#filter
```{r}
#get clr transormed data
phyloseq%>%
  subset_samples(stress %in% "O2")%>%
  physeq_glom_rename(taxrank = "Genus", 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt() ->df_abund

#get compositional data#########################
phyloseq%>%
  subset_samples(stress %in% "O2")%>%
  physeq_glom_rename(taxrank = "Genus", 
                     speedyseq = TRUE)%>%
   microbiome::transform("compositional") %>% 
   psmelt()->df_abund_compositional_genus
#mean abundance to filter 
df_abund_compositional_mean<-setNames(aggregate(df_abund_compositional_genus$Abundance, by=list(df_abund_compositional_genus$Genus), FUN="mean"), c("Genus", "mean_abund"))
df_abund_compositional_mean %>%
  subset(mean_abund >0.0025 )->abundance_filter
#filter clr abundace df based on relative abundace

df_abund %>%
  subset(Genus %in% abundance_filter[,c("Genus")])->clr_O2
#################################################
```

##calculate clr-differences per donor
```{r}
clr_diff<-setNames(as.data.frame(matrix(0,0,5)), c("diff", "Genus", "stress_level", "donor_name", "time"))

stress_level_cond <-c("control", "low","median", "high", "max")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_O2$donor_name)
donor_fam<-unique(clr_O2$Genus)


i=1
for (q in 1: length(time_cond)){
  clr_O2%>%
    subset(time %in% time_cond[q])->sub1a
  
for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% stress_level_cond[m])->sub1
  
  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2
    
    for (k in 1:length(donor_fam)){
    sub2%>%
    subset(Genus %in% donor_fam[k])->sub3
  
      if (nrow(sub3) > 0){
      
      sub3 %>%
          subset(incubation %in% "aerobe")->aerobe
       sub3 %>%
          subset(incubation %in% "anaerobe")->anaerobe
       
      clr_diff[i, c("diff")] <- aerobe$Abundance - anaerobe$Abundance 
      clr_diff[i, c("Genus")] <-donor_fam[k]
      clr_diff[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff[i, c("donor_name")] <-donor_cond[j]
      clr_diff[i, c("time")] <-time_cond[q]
  i=i+1}
    }}}}
```

#add the stats
```{r}
clr_diff<-left_join(clr_diff, mean_abund)
stats_o2<-setNames(all_oxygen_genus[, c("we.ep", "V14", "V15", "Family", "diff.btw")], c("p", "stress_level", "time", "Genus", "diff_aldex"))
stats_o2$Genus<-str_remove(stats_o2$Genus, "Genus:")

for (i in 1:nrow(stats_o2)){
  if (stats_o2[i,c("p")] <0.05 & stats_o2[i,]$p >0.01){stats_o2[i,c("sig")] = "*"}
  if (stats_o2[i,c("p")] <0.01 & stats_o2[i,]$p >0.001){stats_o2[i,c("sig")] = "**"}
  if (stats_o2[i,c("p")] <0.001 ){stats_o2[i,c("sig")] = "***"}
}

diff_stats_o2<-join(clr_diff, stats_o2)
```
#plot
```{r}
diff_stats_o2$stress_level<-factor(diff_stats_o2$stress_level, levels=c("control", "low","median", "high", "max"))
diff_stats_o2$time<-factor(diff_stats_o2$time, levels=c("stress", "post_stress_1","post_stress_2"))


names<-list("stress"= "Stress","post_stress_1"= "Post 1", "post_stress_2"= "Post 2")

facet_labeller <- function(variable,value){
  return(names[value])
}

###betrag von clr difference ausrechnen####
mean_diff<-setNames(aggregate(diff_stats_o2$diff, by=list(diff_stats_o2$Genus, diff_stats_o2$stress_level, diff_stats_o2$time), FUN="median"), c("Genus", "stress_level", "time", "mean_diff"))
mean_diff%>%
  subset(stress_level %in% "max") %>%
  subset(time %in% "stress")->mean_diff
mean_diff$mean_diff<-abs(mean_diff$mean_diff)
######

diff_stats_o2<-left_join(diff_stats_o2, mean_diff[, c("Genus", "mean_diff")])

diff_stats_o2%>%
  # subset(stress_level %in% c("control", "median","max")) %>%
  subset(!donor_name %in% c("D1"))->sub

sub$diff_aldex<-sub$diff_aldex*(-1)

sub %>%
   mutate(Genus = fct_reorder(Genus, sub$abund))%>%
  ggplot(aes(x=diff, y=Genus))+
    # scale_fill_manual(values=c("steelblue4", "coral2"  , "coral4"), labels=c("minimal", "median","max"))+
  geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
  # geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_point(aes(x=diff_aldex, y=Genus, group=stress_level, fill=stress_level ), position = position_dodge(0.82), shape=23, size=3)+
  geom_jitter(aes(group=stress_level, colour=donor_name), position = position_jitterdodge(0.001) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
    ylab("")+xlab("Differences in clr-abundance")+
  labs(colour="Donor", fill= "Stress level")+
 geom_text(data= subset(sub, donor_name %in% c("D7") ), aes(x=diff+2, y=Genus, label=sig, group=stress_level), size=8, colour="black", position = position_dodge2(0.8))+ 
  facet_grid(cols=vars(time), labeller= facet_labeller)+
      ggtitle(expression(O[2]*-specifically~altered~genera))+
  scale_x_continuous(breaks=c(-2.5, 2.5))




ggsave(filename = paste0("clr_dist_genus_without_D1.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/O2/clr_diff",
       width = 28,
       height = 40,
       units = c("cm"))
sub%>%
  subset(donor_name %in% "D5") %>%
  subset(!sig %in% NA)

diff_stats_o2 %>%
  subset(Genus %in% "Blautia")
```

```{r}
diff_stats_o2_fact<-diff_stats_o2

diff_stats_o2_fact$stress_level<-as.numeric(diff_stats_o2_fact$stress_level)


m<-lm(formula = diff ~ stress_level, data = diff_stats_o2_fact%>%subset(Genus %in% "Bacteroides" & time %in% "stress"))
summary(m)

diff_stats_o2_fact %>%
  mutate(diff_aldex = diff_aldex * (-1)) %>%
  subset(Genus %in% "Bacteroides")%>%
  subset(time %in% "stress")%>%
  mutate(predicted = predict(m, diff_stats_o2_fact%>%subset(Genus %in% "Bacteroides" & time %in% "stress"))) %>%
  ggplot(aes(y=diff, x=stress_level))+
    # scale_fill_manual(values=c("steelblue4", "coral2"  , "coral4"), labels=c("minimal", "median","max"))+
  # geom_hline(yintercept = 0, linetype = "solid", colour = "grey", size=1)+
  # geom_hline(yintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  # geom_hline(yintercept = -1, linetype = "dashed", colour = "grey", size=1)+
  # geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_point(aes(y=diff_aldex, x=stress_level), position = position_dodge(0.82), shape=23, size=3)+
  geom_jitter(aes(colour=donor_name), position = position_jitterdodge(0.001) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
    ylab("")+xlab("Differences in clr-abundance")+
  labs(colour="Donor", fill= "Stress level")+
 # geom_text(data= subset(sub, donor_name %in% c("D7") ), aes(x=diff+2, y=Genus, label=sig, group=stress_level), size=8, colour="black", position = position_dodge2(0.8))
  # facet_grid(rows=vars(time), labeller= facet_labeller)+
      ggtitle(expression(O[2]*-specifically~altered~genera))+
  scale_y_continuous(breaks=c(-2.5, 2.5))+
  geom_smooth(aes(y=predicted))

```



###Species-level
```{r}
names(all_oxygen_species)[c(14, 15, 16)]<-c("stress_level", "time", "Species")

#get compositional data#########################
phyloseq%>%
  subset_samples(stress %in% "O2")%>%
  physeq_glom_rename(taxrank = "Species", 
                     speedyseq = TRUE)%>%
   microbiome::transform("compositional") %>% 
   psmelt()->df_abund_compositional_species
#mean abundance to filter 
df_abund_compositional_mean<-setNames(aggregate(df_abund_compositional_species$Abundance, by=list(df_abund_compositional_species$Species), FUN="mean"), c("Species", "mean_abund"))
df_abund_compositional_mean %>%
  subset(mean_abund >0.0025 )->abundance_filter
#filter clr abundace df based on relative abundace

all_oxygen_species %>%
  subset(Species %in% abundance_filter[,c("Species")])->all_oxygen_species_filtered

#################################################
#add the stats


for (i in 1:nrow(all_oxygen_species_filtered)){
  if (all_oxygen_species_filtered[i,c("we.ep")] <0.05 & all_oxygen_species_filtered[i,]$we.ep >0.01){all_oxygen_species_filtered[i,c("sig")] = "*"}
  if (all_oxygen_species_filtered[i,c("we.ep")] <0.01 & all_oxygen_species_filtered[i,]$we.ep >0.001){all_oxygen_species_filtered[i,c("sig")] = "**"}
  if (all_oxygen_species_filtered[i,c("we.ep")] <0.001 ){all_oxygen_species_filtered[i,c("sig")] = "***"}
}


###

all_oxygen_species_filtered$time<-factor(all_oxygen_species_filtered$time, levels= c("stress", "post_stress_1", "post_stress_2"))


all_oxygen_species_filtered$diff.btw <- all_oxygen_species_filterd$diff.btw * (-1)

all_oxygen_species_filtered %>%
  subset(stress_level %in% c("control", "median", "max"))->sub


sub$stress_level<-factor(sub$stress_level, levels= c("control", "median", "max"))

sub%>%
   ggplot(aes(x=diff.btw, y=Species))+
    scale_fill_manual(values=c("steelblue4", "coral2"  , "coral4"), labels=c("minimal", "median","max"))+
  geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
  # geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_point(aes( group=stress_level, fill=stress_level ), position = position_dodge(0.82), shape=23, size=3)+
  # geom_jitter(aes(group=stress_level, colour=donor_name), position = position_jitterdodge(0.001) ,size=1, alpha=0.75)+
  #  # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
  #  #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
  #  #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
    ylab("")+xlab("Differences in clr-abundance")+
  labs(colour="Donor", fill= "Stress level")+
  facet_grid(cols=vars(time), labeller= facet_labeller)+
      ggtitle(expression(O[2]*-specifically~altered~genera))+
  scale_x_continuous(breaks=c(-2.5, 2.5))+
  geom_text( aes(x=diff.btw, y=Species, label=sig, group=stress_level), size=8, colour="black", position = position_dodge2(0.8))


ggsave(filename = paste0("clr_dist_species.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/O2/clr_diff",
       width = 28,
       height = 80,
       units = c("cm"))
```


#H2O2
##Family level
```{r}
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = "Family",
                     speedyseq = TRUE)%>%
   microbiome::transform("clr") %>%
   psmelt() ->df_abund

clr_H2O2<-left_join(df_abund, as.data.frame(phyloseq@tax_table))

mean_abund<-setNames(aggregate(clr_H2O2$Abundance, by=list(clr_H2O2$Family), FUN = "mean"), c("Family", "abund"))

clr_H2O2<-left_join(clr_H2O2, mean_abund)

mean_abund$Family<-as.factor(mean_abund$Family)

clr_H2O2%>%
  mutate(Family = fct_reorder(Family, clr_H2O2$abund)) ->hoi

ordered<-unique(as.character(hoi$Family))[1:15]
# clr_drug$treatment<-factor(clr_drug$treatment, levels=c("DMSO", "omeprazole",  "ciprofloxacin","b_5-FU","a_control"))

clr_H2O2%>%
  subset(Family %in% c(ordered))->clr_H2O2

clr_diff_H2O2<-setNames(as.data.frame(matrix(0,0,5)), c("diff", "Family", "stress_level", "donor_name", "time"))

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_H2O2$donor_name)
donor_fam<-unique(clr_H2O2$Family)


i=1
for (q in 1: length(time_cond)){
  clr_H2O2%>%
    subset(time %in% time_cond[q])->sub1a

for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% c(stress_level_cond[m]))->sub1

  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2

    for (k in 1:length(donor_fam)){
    sub2%>%
    subset(Family %in% donor_fam[k])->sub3

      if (nrow(sub3) > 0){

clr_H2O2%>%
          subset(time %in% time_cond[q] & stress_level %in% c("control") & donor_name %in% donor_cond[j] & Family %in% donor_fam[k])->hoi

      clr_diff_H2O2[i, c("diff")] <- sub3$Abundance - hoi$Abundance


      clr_diff_H2O2[i, c("Family")] <-donor_fam[k]
      clr_diff_H2O2[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff_H2O2[i, c("donor_name")] <-donor_cond[j]
      clr_diff_H2O2[i, c("time")] <-time_cond[q]


  i=i+1}
    }}}}


clr_diff_H2O2<-left_join(clr_diff_H2O2, mean_abund)


stats<-setNames(all_h2o2[, c("we.ep", "V14", "V15", "Family")], c("p", "stress_level", "time", "Family"))

stats$Family<-str_remove(stats$Family, "Family:")

for (i in 1:nrow(stats)){
  if (stats[i,c("p")] <0.05 & stats[i,]$p >0.01){stats[i,c("sig")] = "*"}
  if (stats[i,c("p")] <0.01 & stats[i,]$p >0.001){stats[i,c("sig")] = "**"}
  if (stats[i,c("p")] <0.001 ){stats[i,c("sig")] = "***"}
}



diff_stats<-join(clr_diff_H2O2, stats)

diff_stats$stress_level<-factor(diff_stats$stress_level, levels=c("control", "low","median", "high", "max"))
diff_stats$time<-factor(diff_stats$time, levels=c("stress", "post_stress_1","post_stress_2"))
diff_stats%>%
  mutate(Family = fct_reorder(Family, diff_stats$abund))%>%
  ggplot(aes(x=diff, y=Family))+
  geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
   geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_jitter(aes(group=stress_level, colour=donor_name), position = position_jitterdodge(0.1) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"),
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"),
              axis.title.y = element_text(size=20),
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20),
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
      ylab("")+xlab("Differences in clr-abundance")+labs(colour="Donor")+
 geom_text(data= subset(diff_stats, donor_name %in% c("D1")), aes(label=sig, group=stress_level), position = position_dodge(width=0.75), size=5, colour="red")+ facet_grid(cols=vars(time))+

      ggtitle("Drug-specifically altered families")

ggsave(filename = paste0("clr_dist.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/H2O2/clr_diff",
       width = 25,
       height = 30,
       units = c("cm"))

51.7/14*27
65.3/17*30
```

## Genus level
#filter

```{r}
#get clr transormed data
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = "Genus", 
                     speedyseq = T)%>%
   microbiome::transform("clr") %>% 
   psmelt() ->df_abund

#get compositional data#########################
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = "Genus", 
                     speedyseq = TRUE)%>%
   microbiome::transform("compositional") %>% 
   psmelt()->df_abund_compositional_genus


#mean abundance to filter 
df_abund_compositional_mean<-setNames(aggregate(df_abund_compositional_genus$Abundance, by=list(df_abund_compositional_genus$Genus), FUN="mean"), c("Genus", "mean_abund"))
df_abund_compositional_mean %>%
  subset(mean_abund >0.001 )->abundance_filter
#filter clr abundace df based on relative abundace
df_abund %>%
  subset(Genus %in% abundance_filter[,c("Genus")])->clr_H2O2
#################################################
```
#calculate clr differences per donor
```{r}
#create new df
clr_diff_H2O2<-setNames(as.data.frame(matrix(0,0,5)), c("diff", "Genus", "stress_level", "donor_name", "time"))

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_H2O2$donor_name)
donor_fam<-unique(clr_H2O2$Genus)

#loop over all conditions
i=1
for (q in 1: length(time_cond)){
  clr_H2O2%>%
    subset(time %in% time_cond[q])->sub1a
  
for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% c(stress_level_cond[m]))->sub1
  
  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2
    
      for (k in 1:length(donor_fam)){
      sub2%>%
      subset(Genus %in% donor_fam[k])->sub3
    
        if (nrow(sub3) > 0){
      
clr_H2O2%>%
          subset(time %in% time_cond[q] & stress_level %in% c("control") & donor_name %in% donor_cond[j] & Genus %in% donor_fam[k])->hoi
      clr_diff_H2O2[i, c("diff")] <- sub3$Abundance - hoi$Abundance 
      clr_diff_H2O2[i, c("Genus")] <-donor_fam[k]
      clr_diff_H2O2[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff_H2O2[i, c("donor_name")] <-donor_cond[j]
      clr_diff_H2O2[i, c("time")] <-time_cond[q]
     
  
  i=i+1}
      }}}}

clr_diff_H2O2<-left_join(clr_diff_H2O2, clr_H2O2[, c("Abundance", "Genus")])

###betrag von clr difference ausrechnen####
mean_diff<-setNames(aggregate(clr_diff_H2O2$diff, by=list(clr_diff_H2O2$Genus, clr_diff_H2O2$stress_level, clr_diff_H2O2$time), FUN="median"), c("Genus", "stress_level", "time", "mean_diff"))

mean_diff%>%
  subset(stress_level %in% "high") %>%
  subset(time %in% "stress")->mean_diff
mean_diff$mean_diff<-abs(mean_diff$mean_diff)
######
```



##Species level
```{r}
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = "Species", 
                     speedyseq = TRUE)%>%
   microbiome::transform("clr") %>% 
   psmelt() ->df_abund

clr_H2O2<-left_join(df_abund, as.data.frame(phyloseq@tax_table))

mean_abund<-setNames(aggregate(clr_H2O2$Abundance, by=list(clr_H2O2$Species), FUN = "mean"), c("Species", "abund"))

clr_H2O2<-left_join(clr_H2O2, mean_abund)

mean_abund$Species<-as.factor(mean_abund$Species)

clr_H2O2%>%
  mutate(Species = fct_reorder(Species, clr_H2O2$abund)) ->hoi

ordered<-unique(as.character(hoi$Species))[1:30]
# clr_drug$treatment<-factor(clr_drug$treatment, levels=c("DMSO", "omeprazole",  "ciprofloxacin","b_5-FU","a_control"))

clr_H2O2%>%
  subset(Species %in% c(ordered))->clr_H2O2

clr_diff_H2O2<-setNames(as.data.frame(matrix(0,0,5)), c("diff", "Species", "stress_level", "donor_name", "time"))

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_H2O2$donor_name)
donor_fam<-unique(clr_H2O2$Species)


i=1
for (q in 1: length(time_cond)){
  clr_H2O2%>%
    subset(time %in% time_cond[q])->sub1a
  
for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% c(stress_level_cond[m]))->sub1
  
  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2
    
    for (k in 1:length(donor_fam)){
    sub2%>%
    subset(Species %in% donor_fam[k])->sub3
  
      if (nrow(sub3) > 0){
      
clr_H2O2%>%
          subset(time %in% time_cond[q] & stress_level %in% c("control") & donor_name %in% donor_cond[j] & Species %in% donor_fam[k])->hoi
        
      clr_diff_H2O2[i, c("diff")] <- sub3$Abundance - hoi$Abundance 
   
      
      clr_diff_H2O2[i, c("Species")] <-donor_fam[k]
      clr_diff_H2O2[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff_H2O2[i, c("donor_name")] <-donor_cond[j]
      clr_diff_H2O2[i, c("time")] <-time_cond[q]
     
  
  i=i+1}
    }}}}


clr_diff_H2O2<-left_join(clr_diff_H2O2, mean_abund)


stats<-setNames(all_h2o2_species[, c("we.ep", "V14", "V15", "Family")], c("p", "stress_level", "time", "Species"))

stats$Species<-str_remove(stats$Species, "Species:")

for (i in 1:nrow(stats)){
  if (stats[i,c("p")] <0.05 & stats[i,]$p >0.01){stats[i,c("sig")] = "*"}
  if (stats[i,c("p")] <0.01 & stats[i,]$p >0.001){stats[i,c("sig")] = "**"}
  if (stats[i,c("p")] <0.001 ){stats[i,c("sig")] = "***"}
}



diff_stats<-join(clr_diff_H2O2, stats)

diff_stats$stress_level<-factor(diff_stats$stress_level, levels=c("control", "low","median", "high", "max"))
diff_stats$time<-factor(diff_stats$time, levels=c("stress", "post_stress_1","post_stress_2"))
diff_stats%>%
  mutate(Species = fct_reorder(Species, diff_stats$abund))%>%
  ggplot(aes(x=diff, y=Species))+
  geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
   geom_boxplot(aes(fill=stress_level), outlier.shape = NA)+
  geom_jitter(aes(group=stress_level, colour=donor_name), position = position_jitterdodge(0.1) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
      ylab("")+xlab("Differences in clr-abundance")+labs(colour="Donor")+
 geom_text(data= subset(diff_stats, donor_name %in% c("D1")), aes(label=sig, group=stress_level), position = position_dodge(width=0.75), size=5, colour="red")+ facet_grid(cols=vars(time))+
  
      ggtitle("Drug-specifically altered species")

ggsave(filename = paste0("clr_dist_species.jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/H2O2/clr_diff",
       width = 40,
       height = 60,
       units = c("cm"))

51.7/14*27
65.3/17*30
```


##Species level
```{r}
tax_level<-c("Lachnospiraceae", "Bacteroidaceae", "Ruminococcaceae")

# tax_level<- c("Bacteroides",  "Blautia", "Anaerostipes", "Ruminococcus", "Faecalibacterium",    "Parabacteroides", "Phascolarctobacterium")
rank="Species"

for (t in 1:length(tax_level)){
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = rank, 
                     speedyseq = TRUE)%>%
   microbiome::transform("clr") %>% 
   psmelt() ->df_abund
  
  phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = rank, 
                     speedyseq = TRUE)%>%
   microbiome::transform("compositional") %>% 
   psmelt() %>%
  subset(Family %in% c(tax_level[t]))->df_abund_compositional_genus


df_abund_compositional_mean<-setNames(aggregate(df_abund_compositional_genus$Abundance, by=list(df_abund_compositional_genus$Species), FUN="mean"), c("Species", "mean_abund"))

df_abund_compositional_mean %>%
  subset(mean_abund >0.005 )->abundance_filter


print(abundance_filter)

# clr_H2O2<-left_join(df_abund, as.data.frame(phyloseq@tax_table))
df_abund%>%
  subset(Family %in% c(tax_level[t]))%>%
  subset(Species %in% c(abundance_filter$Species))->clr_H2O2



# mean_abund<-setNames(aggregate(clr_H2O2$Abundance, by=list(clr_H2O2[,rank]), FUN = "mean"), c(rank, "abund"))
# clr_H2O2<-left_join(clr_H2O2, mean_abund)
# 
# 
# mean_abund[,rank]<-as.factor(mean_abund[,rank])
# 
# clr_H2O2%>%
#   subset(Genus %in% (tax_level[t]))%>%
#   mutate(Strain = fct_reorder(Strain, clr_H2O2$abund)) ->hoi
# 
# ordered<-unique(as.character(hoi[,rank]))[1:30]
# # clr_drug$treatment<-factor(clr_drug$treatment, levels=c("DMSO", "omeprazole",  "ciprofloxacin","b_5-FU","a_control"))
# 
# clr_H2O2%>%
#   subset(Strain %in% c(ordered))->clr_H2O2

clr_diff_H2O2<-setNames(as.data.frame(matrix(0,0,5)), c("diff", rank, "stress_level", "donor_name", "time"))

stress_level_cond <-c( "low","median", "high")
time_cond<-c("stress", "post_stress_1", "post_stress_2")
donor_cond<-unique(clr_H2O2$donor_name)
donor_fam<-unique(clr_H2O2$Species)


i=1
for (q in 1: length(time_cond)){
  clr_H2O2%>%
    subset(time %in% time_cond[q])->sub1a
  
for (m in 1: length(stress_level_cond)){
  sub1a%>%
    subset(stress_level %in% c(stress_level_cond[m]))->sub1
  
  for (j in 1:length(donor_cond)){
    sub1%>%
    subset(donor_name %in% donor_cond[j])->sub2
    
    for (k in 1:length(donor_fam)){
    sub2%>%
    subset(Species %in% donor_fam[k])->sub3
  
      if (nrow(sub3) > 0){
      
clr_H2O2%>%
          subset(time %in% time_cond[q] & stress_level %in% c("control") & donor_name %in% donor_cond[j] & Species %in% donor_fam[k])->hoi
        
      clr_diff_H2O2[i, c("diff")] <- sub3$Abundance - hoi$Abundance 
   
      
      clr_diff_H2O2[i, c(rank)] <-donor_fam[k]
      clr_diff_H2O2[i, c("stress_level")] <-stress_level_cond[m]
      clr_diff_H2O2[i, c("donor_name")] <-donor_cond[j]
      clr_diff_H2O2[i, c("time")] <-time_cond[q]
     
  
  i=i+1}
    }}}}


# clr_diff_H2O2<-left_join(clr_diff_H2O2, mean_abund)
# 
# clr_diff_H2O2%>%
#   subset(!rank %in% paste(tax_level[t], "Genus"))->clr_diff_H2O2
# 
# stats<-setNames(all_h2o2_species[, c("we.ep", "V14", "V15", "Family")], c("p", "stress_level", "time", rank))
# 
# stats$Species<-str_remove(stats$Species, "Species:")
# 
# for (i in 1:nrow(stats)){
#   if (stats[i,c("p")] <0.05 & stats[i,]$p >0.01){stats[i,c("sig")] = "*"}
#   if (stats[i,c("p")] <0.01 & stats[i,]$p >0.001){stats[i,c("sig")] = "**"}
#   if (stats[i,c("p")] <0.001 ){stats[i,c("sig")] = "***"}
# }
# 
x<-tax_level[t]
# 
# diff_stats<-join(clr_diff_H2O2, stats)
clr_diff_H2O2$stress_level<-factor(clr_diff_H2O2$stress_level, levels=c("control", "low","median", "high", "max"))
clr_diff_H2O2$time<-factor(clr_diff_H2O2$time, levels=c("stress", "post_stress_1","post_stress_2"))


  
clr_diff_H2O2%>%  
  # mutate(Species = fct_reorder(Species, diff_stats$abund))%>%
  ggplot(aes(x=diff, y=Species))+
  geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
   geom_boxplot(aes(fill=time), outlier.shape = NA)+
  geom_jitter(aes(group=time, colour=donor_name), position = position_jitterdodge(0.1) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
      ylab("")+xlab("Differences in clr-abundance")+labs(colour="Donor")+
 # geom_text(data= subset(diff_stats, donor_name %in% c("D1")), aes(label=sig, group=stress_level), position = position_dodge(width=0.75), size=5, colour="red")+ 
facet_grid(cols=vars(stress_level))+
  
      ggtitle(paste("H2O2-specifically altered", x))

ggsave(filename = paste0(x, ".jpeg"),
       path = "C:/Users/zuendj/Desktop/03_data R/wp2_complex/output/H2O2/clr_diff/family",
       width = 50,
       height = 80,
       units = c("cm"))}

51.7/14*27
65.3/17*30
```


```{r}
phyloseq%>%
  subset_samples(stress %in% "H2O2")%>%
  physeq_glom_rename(taxrank = "Genus", 
                     speedyseq = TRUE)%>%
   microbiome::transform("compositional") %>% 
   psmelt() %>%
  subset(Genus %in% "Streptococcus")->df_abund



df_abund%>%  
  # mutate(Species = fct_reorder(Species, diff_stats$abund))%>%
  ggplot(aes(x=Abundance, y=time))+
  # geom_vline(xintercept = 0, linetype = "solid", colour = "grey", size=1)+
  # geom_vline(xintercept = 1, linetype = "dashed", colour = "grey", size=1)+
  # geom_vline(xintercept = -1, linetype = "dashed", colour = "grey", size=1)+
   geom_boxplot( outlier.shape = NA)+
  geom_jitter(aes(group=time, colour=donor_name), position = position_jitterdodge(0.1) ,size=1, alpha=0.75)+
   # scale_fill_manual(values = c( "cyan4" , "darkgoldenrod1", "#F8766D"),
   #                      labels = c(  "5-FU","Ciprofloxacin", "Omeprazole"), 
   #                     name="Drug")+
   theme(axis.text.y= element_text(size = 20, face="italic"), 
              axis.title.y = element_text(size=20), 
              axis.title.x = element_text(size=20),
              axis.text.x = element_text(size=20),
              legend.text = element_text(size = 20),
              legend.title = element_text(size= 20),
              plot.title = element_text(size=20), 
         panel.background = element_rect(fill="white"),
        panel.border=element_rect(fill=NA), strip.text = element_text(size=20))+
      ylab("")+xlab("Differences in clr-abundance")+labs(colour="Donor")+
 # geom_text(data= subset(diff_stats, donor_name %in% c("D1")), aes(label=sig, group=stress_level), position = position_dodge(width=0.75), size=5, colour="red")+ 
facet_grid(cols=vars(stress_level))+scale_x_log10()
```



